src/main/scala/org/broadinstitute/dsde/firecloud/model/Profile.scala

```diff
@@ -13,6 +13,10 @@ case class ThurloeKeyValue(
   userId: Option[String] = None,
   keyValuePair: Option[FireCloudKeyValue] = None)
 
+case class ThurloeKeyValues(
```

Comment:
Most of your suggestions have a similar theme so I'll just respond to one of them with my thoughts:

1) I'm not sure why userId or the keyValues are optional. I was just following the already established pattern. I think that it may have something to do with JSON unmarshalling? I'll do some more digging to understand why since "I was just following the pattern" is a crappy answer
2) Most of the terminology used here was chosen to be consistent with Thurloe- to make the mental mapping a bit easier. 
3) I agree with your point about `FireCloudKeyValue`. I think that a rename would be useful there.